drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -780,7 +781,16 @@ end
 
 local function handle_set_color_temperature(driver, device, cmd)
   local endpoint_id = device:component_to_endpoint(cmd.component)
-  local temp_in_mired = utils.round(MIRED_KELVIN_CONVERSION_CONSTANT/cmd.args.temperature)
+  local temp_in_kelvin = cmd.args.temperature
```

Comment:
Can you help my understanding here - shouldn't our boundaries set for the colorTemperature capability prevent us from setting a colorTemp value outside the device's known range? It seems like if we know the colorTemp range of the device, we should be able to prevent this issue via dynamic constraints, unless I am missing something here.

Reply:
The issue is that the dynamic constraints set the min and max values in the plugin using rounded kelvin values (to the nearest 100 I believe). Take the example from the ticket linked to this PR. The temp was set to 6500 in the ST app, which is the maximum of the allowed range. When we convert this to mireds we get 154 (154 mired = 6493 K). However, this max of 6500 was originally calculated from converting 153 mireds to kelvin (153 mireds = 6535 K). So, in order to be able to use the full range allowed by the device, we should send 153 to the device rather than 154 if we set the capability to 6500.